app/scripts/controllers/push-platform-notifications/services/services.ts

```diff
@@ -215,13 +217,25 @@ export async function listenToPushNotifications(
     },
   );
 
+  // Notification Click Listener
+  const notificationClickHandler = (event: NotificationEvent) => {
+    onNotificationClick(event, onNotificationClicked);
+  };
+  sw.addEventListener('notificationclick', notificationClickHandler);
```

Comment:
How does the service worker listen for click events?

Reply:
Service workers can subscribe to the 'notificationclick' listener [[1]](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope/notificationclick_event). This can be used to choose what happens when a user clicks on a push notification (e.g. dismissing and also navigation).